plugins/modules/yum_versionlock.py

```diff
@@ -111,18 +111,26 @@ def get_versionlock_packages(self):
     def ensure_state(self, packages, command):
         """ Ensure packages state """
         rc, out, err = self.module.run_command([self.yum_bin, "-q", "versionlock", command] + packages)
+        # If no package can be found this will be written on stdout with rc 0
+        if 'No package found for' in out:
+            self.module.fail_json(msg=out)
         if rc == 0:
             return True
         self.module.fail_json(msg="Error: " + to_native(err) + to_native(out))
 
 
 def match(entry, name):
+    match = False
     m = NEVRA_RE_YUM.match(entry)
     if not m:
         m = NEVRA_RE_DNF.match(entry)
     if not m:
         return False
-    return fnmatch(m.group("name"), name)
+    if fnmatch(m.group("name"), name):
+        match = True
+    if fnmatch(m.group(), '{0}*'.format(name)):
```

Comment:
Doesn't this mean that if I ask to lock a package called `foo`, that it will also match a package called `foobar`? Using `'{0}-*` has a similar problem, `foo` will also match `foo-bar`. (Assuming `-` is allowed in package names. I don't use yum/dnf based systems, so :shrug:.)

Also your fix only seems to work with yum, but not with dnf, according to https://github.com/ansible-collections/community.general/blob/b9b0ea855abbd1c239688f582caafd95e82b48c5/plugins/modules/yum_versionlock.py#L90-L93.

Reply:
I tested this on CentOS 7 with legacy yum as well as with CentOS Stream where yum is just an alias for dnf.